src/core/model/table/dynamic_table_data.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "config/exceptions.h"
+#include "config/tabular_data/input_table_type.h"
+
+namespace model {
+
+struct DynamicTableData {
+private:
+    std::vector<std::vector<std::string>> full_table_;
+    std::unordered_set<size_t> deleted_rows_{};
+
+public:
+    DynamicTableData(config::InputTable input_table) {
+        full_table_.resize(input_table->GetNumberOfColumns());
+        while (input_table->HasNextRow()) {
+            std::vector<std::string> row = input_table->GetNextRow();
+            for (size_t i = 0; i < row.size(); ++i) {
+                full_table_[i].emplace_back(std::move(row[i]));
+            }
+        };
+    }
+
+    size_t GetNumRowsActual() const {
+        return full_table_[0].size() - deleted_rows_.size();
+    }
+
+    size_t GetNumRowsTotal() const {
+        return full_table_[0].size();
+    }
+
+    std::string const& GetValue(size_t row_index, size_t col_index) const {
+        assert(col_index < full_table_.size() && row_index < full_table_[col_index].size());
+        return full_table_[col_index][row_index];
+    }
+
+    void Update(config::InputTable insert_data, config::InputTable update_data,
+                std::unordered_set<size_t> const& delete_data) {
+        for (size_t row_id : delete_data) {
+            for (size_t i = 0; i < full_table_.size(); ++i) {
+                full_table_[i][row_id].clear();
+            }
+            deleted_rows_.emplace(row_id);
+        }
+        if (insert_data != nullptr) {
+            while (insert_data->HasNextRow()) {
+                std::vector<std::string> row = insert_data->GetNextRow();
+                for (size_t i = 0; i < row.size(); ++i) {
+                    full_table_[i].emplace_back(std::move(row[i]));
```

Comment:
`emplace_back` doesn't do anything special for a move constructor.
```suggestion
                    full_table_[i].push_back(std::move(row[i]));
```